pkg/subnet/local.go

```diff
@@ -280,47 +333,56 @@ func (d *SubnetDeployer) setupLocalEnv() (string, error) {
 			return "", err
 		}
 	}
+	ux.Logger.PrintToUser("Avalanchego installation successful")
 	return filepath.Join(binDir, avagoSubDir), nil
 }
 
 // WaitForHealthy polls continuously until the network is ready to be used
-func (d *SubnetDeployer) WaitForHealthy(ctx context.Context, cli client.Client, healthCheckInterval time.Duration) ([]string, error) {
+func (d *SubnetDeployer) WaitForHealthy(
+	ctx context.Context,
+	cli client.Client,
+	healthCheckInterval time.Duration,
+) ([]string, []string, []string, error) {
 	cancel := make(chan struct{})
 	go ux.PrintWait(cancel)
 	for {
 		select {
 		case <-ctx.Done():
 			cancel <- struct{}{}
-			return nil, ctx.Err()
+			return nil, nil, nil, ctx.Err()
 		case <-time.After(healthCheckInterval):
 			cancel <- struct{}{}
 			d.log.Debug("polling for health...")
 			go ux.PrintWait(cancel)
 			resp, err := cli.Health(ctx)
 			if err != nil {
-				return nil, fmt.Errorf("the health check failed to complete. The server might be down or have crashed, check the logs! %s", err)
+				return nil, nil, nil, fmt.Errorf("the health check failed to complete. The server might be down or have crashed, check the logs! %s", err)
 			}
 			if resp.ClusterInfo == nil {
 				d.log.Debug("warning: ClusterInfo is nil. trying again...")
 				continue
 			}
-			if len(resp.ClusterInfo.CustomVms) == 0 {
-				d.log.Debug("network is up but custom VMs are not installed yet. polling again...")
+			if !resp.ClusterInfo.Healthy {
+				d.log.Debug("network is not healthy. polling again...")
 				continue
 			}
 			if !resp.ClusterInfo.CustomVmsHealthy {
 				d.log.Debug("network is up but custom VMs are not healthy. polling again...")
 				continue
 			}
+			d.log.Debug("network is up and custom VMs are up")
+			blockchains := []string{}
+			for vmID, vmInfo := range resp.ClusterInfo.CustomVms {
```

Comment:
I wonder if we are overloading the Health check here. What do you think of storing this into the `d *SubnetDeployer` struct for example, or some other means? `WaitForHealthy` would then really just do that, and the storing of data (for later printing) would be done elsewhere?

Reply:
Ok maybe WaitForHealthy can just return ClusterInfo, and the caller creates the other stuff